libcxx/test/libcxx/ranges/range.adaptors/range.concat/iterator.valueless_by_exception.pass.cpp

```diff
@@ -0,0 +1,197 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// REQUIRES: std-at-least-c++26
+// UNSUPPORTED: no-exceptions
+
+#include <iostream>
+#include <ranges>
+#include <utility>
+#include <vector>
+
+#include "check_assertion.h"
+#include "double_move_tracker.h"
+#include "test_iterators.h"
+
+int globalArray[8] = {0, 1, 2, 3, 4, 5, 6, 7};
+
+
+template <class It, class ItTraits = It >
+class ThrowOnMoveIterator
+{
+    typedef std::iterator_traits<ItTraits> Traits;
+    It it_;
+    support::double_move_tracker tracker_;
+    bool moved = false; 
+
+    template <class U, class T> friend class ThrowOnMoveIterator;
+public:
+    typedef          std::input_iterator_tag                   iterator_category;
+    typedef typename Traits::value_type                        value_type;
+    typedef typename Traits::difference_type                   difference_type;
+    typedef It                                                 pointer;
+    typedef typename Traits::reference                         reference;
```

Comment:
We should consistently use `using`. Ditto `libcxx/test/std/ranges/range.adaptors/range.concat/iterator/ctor.pass.cpp`.